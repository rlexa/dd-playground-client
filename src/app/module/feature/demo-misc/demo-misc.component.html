<div ddFlexbox="vyt">
  <mat-card appearance="outlined" class="childv">
    <mat-card-title>Graphsky</mat-card-title>
    <mat-card-content>
      <div ddFlexbox="hysw">
        <div ddFlexbox="vys">
          <div ddFlexbox="hyc">
            <button mat-button color="accent" (click)="graphskyAddTestData()">+ TEST</button>
            <button mat-button color="warn" (click)="graphskyDrop()">DROP</button>
          </div>
          <app-simple-view
            class="hovered-elevated"
            subheader="Stats"
            [data]="graphskyLog$ | async"
            [isWidthConstrained]="false"
            [isDense]="true"
          />
        </div>

        <app-simple-view
          class="hovered-elevated"
          subheader="All"
          [data]="graphskyAll$ | async"
          [isWidthConstrained]="false"
          [isDense]="true"
        />
        <app-simple-view
          class="hovered-elevated"
          subheader="Likes"
          [data]="graphskyLiked$ | async"
          [isWidthConstrained]="false"
          [isDense]="true"
        />
        <app-simple-view
          class="hovered-elevated"
          subheader="Diversity"
          [data]="graphskyDiversity$ | async"
          [isWidthConstrained]="false"
          [isDense]="true"
        />
      </div>
    </mat-card-content>
  </mat-card>
</div>
