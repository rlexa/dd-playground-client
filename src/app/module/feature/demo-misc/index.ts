export * from './demo-misc.component';
