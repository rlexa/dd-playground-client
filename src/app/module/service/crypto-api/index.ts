export * from './crypto-api.module';
export * from './crypto-api.service';
