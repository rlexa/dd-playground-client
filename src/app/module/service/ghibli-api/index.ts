export * from './ghibli-api.service';
