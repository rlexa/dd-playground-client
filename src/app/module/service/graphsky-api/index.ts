export { IGraphskyData, IGraphskyLinkRequest, IGraphskyNode } from './graphsky';
export * from './graphsky-api.module';
export * from './graphsky.service';
