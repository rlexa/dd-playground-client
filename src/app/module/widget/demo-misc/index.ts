export * from './demo-misc.component';
export * from './demo-misc.module';
