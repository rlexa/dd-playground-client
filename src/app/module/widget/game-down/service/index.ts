export * from './game-down.service';
