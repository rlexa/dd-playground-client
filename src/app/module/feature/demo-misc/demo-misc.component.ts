import {CommonModule} from '@angular/common';
import {ChangeDetectionStrategy, Component, inject} from '@angular/core';
import {MatButtonModule} from '@angular/material/button';
import {MatCardModule} from '@angular/material/card';
import {map} from 'rxjs/operators';
import {FlexboxDirective} from '../../directive/flexbox';
import {GraphskyService} from '../../service/graphsky-api';
import {SimpleViewComponent} from '../../widget/simple-view';

@Component({
  selector: 'app-demo-misc',
  templateUrl: './demo-misc.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
  standalone: true,
  imports: [CommonModule, MatButtonModule, MatCardModule, FlexboxDirective, SimpleViewComponent],
  providers: [GraphskyService],
})
export class DemoMiscComponent {
  private readonly graphsky = inject(GraphskyService);

  readonly graphskyLog$ = this.graphsky.log$;
  readonly graphskyNodes$ = this.graphsky.nodeCount$;
  readonly graphskyAll$ = this.graphsky.change$.pipe(
    map(() =>
      this.graphsky.query((nodes, _) =>
        nodes
          .filter((node) => 'name' in node.data)
          .sort((aa, bb) => aa.data.name.toString().localeCompare(bb.data.name.toString()))
          .reduce(
            (acc, node) => ({
              ...acc,
              [node.data.name.toString() + ' ' + node.data.sex.toString()]: node.out
                .map((link) => link.data.relationship + ' ' + link.to.data.name)
                .join(', '),
            }),
            {},
          ),
      ),
    ),
  );
  readonly graphskyLiked$ = this.graphsky.change$.pipe(
    map(() =>
      this.graphsky.query((_, links) =>
        links
          .filter((link) => link.data.relationship === 'likes')
          .map((link) => link.to)
          .sort((aa, bb) => aa.data.name.toString().localeCompare(bb.data.name.toString()))
          .reduce(
            (acc, node) => ({
              ...acc,
              [node.data.name.toString()]: (acc[node.data.name.toString()] || 0) + 1,
            }),
            {},
          ),
      ),
    ),
  );
  readonly graphskyDiversity$ = this.graphsky.change$.pipe(
    map(() =>
      this.graphsky.query((nodes, _) =>
        nodes
          .filter((node) => 'sex' in node.data)
          .reduce(
            (acc, node) => ({
              ...acc,
              [node.data.sex.toString()]: (acc[node.data.sex.toString()] || 0) + 1,
            }),
            {},
          ),
      ),
    ),
  );

  graphskyAddTestData = () => {
    this.graphsky.add([
      {name: 'Alice', sex: 'f'},
      {name: 'Bob', sex: 'm'},
      {name: 'Clark', sex: 'm'},
      {name: 'Donnie', sex: 'm'},
    ]);
    this.graphsky.link([
      {fromNotExact: true, toNotExact: true, data: {relationship: 'likes'}, from: {name: 'Alice'}, to: {name: 'Bob'}},
      {fromNotExact: true, toNotExact: true, data: {relationship: 'likes'}, from: {name: 'Alice'}, to: {name: 'Clark'}},
      {fromNotExact: true, toNotExact: true, data: {relationship: 'likes'}, from: {name: 'Bob'}, to: {name: 'Clark'}},
      {fromNotExact: true, toNotExact: true, data: {relationship: 'likes'}, from: {name: 'Clark'}, to: {name: 'Donnie'}},
      {fromNotExact: true, toNotExact: true, data: {relationship: 'likes'}, from: {name: 'Donnie'}, to: {name: 'Alice'}},
    ]);
  };

  graphskyDrop = () => this.graphsky.drop();
}
