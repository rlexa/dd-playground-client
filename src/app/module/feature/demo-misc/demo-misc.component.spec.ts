import {ComponentFixture} from '@angular/core/testing';
import {MockBuilder, MockRender} from 'ng-mocks';
import {GraphskyService} from '../../service/graphsky-api';
import {DemoMiscComponent} from './demo-misc.component';

describe('DemoMiscComponent', () => {
  let fixture: ComponentFixture<DemoMiscComponent>;

  beforeEach(() =>
    MockBuilder(DemoMiscComponent).beforeCompileComponents((tb) =>
      tb.overrideComponent(DemoMiscComponent, {set: {providers: [GraphskyService]}}),
    ),
  );

  beforeEach(() => {
    fixture = MockRender(DemoMiscComponent);
    fixture.detectChanges();
  });

  test('renders', () => expect(fixture).toMatchSnapshot());
});
