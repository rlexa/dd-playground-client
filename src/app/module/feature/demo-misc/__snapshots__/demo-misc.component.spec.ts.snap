// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoMiscComponent renders 1`] = `
<mock-render>
  <app-demo-misc>
    <div
      ddflexbox="vyt"
      ng-reflect-dd-flexbox="vyt"
    >
      <mat-card
        appearance="outlined"
        class="childv"
        ng-reflect-appearance="outlined"
      >
        <mat-card-title>
          Graphsky
        </mat-card-title>
        <mat-card-content>
          <div
            ddflexbox="hysw"
            ng-reflect-dd-flexbox="hysw"
          >
            <div
              ddflexbox="vys"
              ng-reflect-dd-flexbox="vys"
            >
              <div
                ddflexbox="hyc"
                ng-reflect-dd-flexbox="hyc"
              >
                <button
                  color="accent"
                  mat-button=""
                  ng-reflect-color="accent"
                >
                  + TEST
                </button>
                <button
                  color="warn"
                  mat-button=""
                  ng-reflect-color="warn"
                >
                  DROP
                </button>
              </div>
              <app-simple-view
                class="hovered-elevated"
                ng-reflect-data="[object Object]"
                ng-reflect-is-dense="true"
                ng-reflect-is-width-constrained="false"
                ng-reflect-subheader="Stats"
                subheader="Stats"
              />
            </div>
            <app-simple-view
              class="hovered-elevated"
              ng-reflect-data="[object Object]"
              ng-reflect-is-dense="true"
              ng-reflect-is-width-constrained="false"
              ng-reflect-subheader="All"
              subheader="All"
            />
            <app-simple-view
              class="hovered-elevated"
              ng-reflect-data="[object Object]"
              ng-reflect-is-dense="true"
              ng-reflect-is-width-constrained="false"
              ng-reflect-subheader="Likes"
              subheader="Likes"
            />
            <app-simple-view
              class="hovered-elevated"
              ng-reflect-data="[object Object]"
              ng-reflect-is-dense="true"
              ng-reflect-is-width-constrained="false"
              ng-reflect-subheader="Diversity"
              subheader="Diversity"
            />
          </div>
        </mat-card-content>
      </mat-card>
    </div>
  </app-demo-misc>
</mock-render>
`;
